### PR TITLE
Remove most `MapDomain.Groupable`

### LINKED ISSUE #1097: Remove `MapDomain.Groupable`
`MapDomain` keys require more than just a `Printable.S`, which is constantly annoying:
https://github.com/goblint/analyzer/blob/a651cccf69af65581222d7e8e046b8ad67510cf4/src/domains/mapDomain.ml#L58-L65
AFAIK only base analysis' `CPA` uses non-trivial grouping and just in its debug `pretty` printing (not even in XML), so the grouped printing code could just be specialized in `CPA` while all other uses of `MapDomain` avoid the extra complication.

### PR DESCRIPTION
**Pull request by sim642**
Closes #1097. The interface is kept such that base's `CPA` could still explicitly use grouped pretty-printing.

### CODE REVIEW

File: src/cdomains/baseDomain.ml [hunk 1]
```diff
@@ -6,20 +6,14 @@ module BI = IntOps.BigIntOps
 
 module CPA =
 struct
+  module M0 = MapDomain.MapBot (Basetype.Variables) (VD)
```
Comment (michael-schwarz) [inline src/cdomains/baseDomain.ml:9]: Why did you remove `HashCashed` here?

**Review by michael-schwarz — APPROVED** [review]
Once my comment about HashCached is addressed, this can be merged.

File: src/cdomains/baseDomain.ml (hunk 1 above)
Comment (sim642) [reply src/cdomains/baseDomain.ml:9]: It's still there below on line 15. This is just the innermost `MapDomain` whose printing is changed to preserve the old grouping. After that it's wrapped the same way.

Comment (michael-schwarz) [reply src/cdomains/baseDomain.ml:9]: But the wrapping in line 15 only applies to `Glob`, not the `CPA` itself. Or am I missing something?

Comment (michael-schwarz) [reply src/cdomains/baseDomain.ml:9]: ah, nvm the diff is just a bit confusing. sorry about that

Comment (sim642) [reply src/cdomains/baseDomain.ml:9]: There's no `Glob`, it was just some old unused module that I removed.

Comment (michael-schwarz) [reply src/cdomains/baseDomain.ml:9]: Yeah, sorry. It's good to merge then.
